unified-codes/tools/scripts/dgraph/migration/v2/import.js

```diff
@@ -0,0 +1,294 @@
+const excel = require('exceljs');
+const dgraph = require('dgraph-js');
+
+const inputFilename = process.argv[2] || './UC Spreadsheet 3.0.xlsm';
+
+const clientStub = new dgraph.DgraphClientStub(`localhost:9080`);
+const dgraphClient = new dgraph.DgraphClient(clientStub);
+
+const combinationDrugs = [];
+
+const insertRootNodes = async () => {
+  const query = `query {
+    DrugCategory as var(func: eq(dgraph.type, Category)) @filter(eq(name, "Drug"))
+    ConsumableCategory as var(func: eq(dgraph.type, Category)) @filter(eq(name, "Consumable"))
+    OtherCategory as var(func: eq(dgraph.type, Category)) @filter(eq(name, "Other"))
+  }`;
+
+  const mutation = new dgraph.Mutation();
+  mutation.setSetNquads(`
+    uid(DrugCategory) <name> "Drug" .
+    uid(ConsumableCategory) <name> "Consumable" .
+    uid(OtherCategory) <name> "Other" .
+    uid(DrugCategory) <dgraph.type> "Category" .
+    uid(ConsumableCategory) <dgraph.type> "Category" .
+    uid(OtherCategory) <dgraph.type> "Category" .
+    uid(DrugCategory) <code> "933f3f00" .
+    uid(ConsumableCategory) <code> "77fcbb00" .
+  `);
+  // TODO: Source category codes from config or spreadsheet rather than hardcoding
+
+  await processDgraphUpsert(query, mutation);
+};
+
+const processRow = async (row) => {
+  const productName = row.getCell(1).value;
+  const synonyms = row.getCell(2).value;
+  const combinations = row.getCell(3).value;
+  const routeName = row.getCell(4).value;
+  const doseFormName = row.getCell(5).value;
+  const doseQualificationName = row.getCell(6).value;
+  const doseStrengthName = row.getCell(7).value;
+  const doseUnitName = row.getCell(8).value;
+  const immediatePackagingName = row.getCell(9).value;
+  const packSizeName = row.getCell(10).value;
+  const outerPackagingName = row.getCell(11).value;
+
+  const categoryCode = row.getCell(12).value;
+  const productCode = row.getCell(13).value;
+  const routeCode = row.getCell(14).value;
+  const doseFormCode = row.getCell(15).value;
+  const doseQualificationCode = row.getCell(16).value;
+  const doseStrengthCode = row.getCell(17).value;
+  const doseUnitCode = row.getCell(18).value;
+  const immediatePackagingCode = row.getCell(19).value;
+  const packSizeCode = row.getCell(20).value;
+  const outerPackagingCode = row.getCell(21).value;
+
+  // Properties
+  const gs1 = row.getCell(22).value;
+  const rxNavUC2 = row.getCell(23).value;
+  const rxNavUC67 = row.getCell(24).value;
+  const atcUC2 = row.getCell(25).value;
+  const dddUC2 = row.getCell(26).value;
+  const atcUC5 = row.getCell(27).value;
+  const dddUC5 = row.getCell(28).value;
+  const emlUC2 = row.getCell(29).value;
+  const emlUC67 = row.getCell(30).value;
+  const unspsc = row.getCell(31).value;
+  const nzulmUC2 = row.getCell(32).value;
+  const nzulmUC67 = row.getCell(33).value;
+  const snomed = row.getCell(34).value;
+  const drugbank = row.getCell(35).value;
+  const usfda = row.getCell(36).value;
```

Comment:
I think could probably make this a bit fancy with some functional stuff, e.g.

```
propertyCells = {
  gs1: 22,
  rvNavUC2: 23,
  ...
};

const properties = Object.values(cellProperties).reduce( .... );
```

Would then end up with an object like:

```
{
  gs1: "12345678",
  rxNavUC2: "987654321",
  ...
}
```

where `"123456789"` etc. are the values taken from the spreadsheet.

Again, not urgent... something which can be refined later!

![smile](https://user-images.githubusercontent.com/43223637/99206427-3c42b000-2820-11eb-9fbd-7b2d5e94e4bf.gif)